subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/SymbolExtractorOsConfig.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.toolchain.internal;
+
+import com.google.common.collect.Lists;
+import org.gradle.internal.os.OperatingSystem;
+
+import java.util.List;
+
+public enum SymbolExtractorOsConfig {
+    OBJCOPY("objcopy", Lists.newArrayList("--only-keep-debug"), ".debug"),
+    DSYMUTIL("dsymutil", Lists.<String>newArrayList("-f"), ".dwarf") {
```

Comment:
According to Xcode build process, the command ran after the build on the binary during a release build is `dsymutil App -o App.dSYM`. This is pretty close to what you have here. The only difference is Xcode generate a bundle version of the `.dSYM`. The comment is we should use `.dSYM` instead of `.dwarf` as the file extension.

Reply:
So `dsymutil` can produce either a flat file or a directory.  If you don't provide it with a destination it will produce a '.dwarf' file if creating a flat file or a '.dSYM' directory if generating a directory.  I simply followed that pattern.  There's perhaps a discussion here as to whether we should instead produce the directory structure instead of the flat file, but I think if we stick with the flat file, a '.dwarf' extension makes the most sense.